PHP 8.2 fixes

cedric-anne opened the pull request:
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Each commit should be reviewed individually.

See https://php.watch/versions/8.2

1. Usage of dynamic properties is deprecated. It means that using `$this->myvar = '';` when `$myvar` is not declared in class properties will trigger a deprecation error.
To fix this, I used different solutions.
 - I replaced class properties by local variables when usage of properties was not relevant.
 - I declared properties with a private visibility when they were used only locally.
 - I declared properties with a public visibility when they were used by other core or plugin classes.

2. Usage of some "pseudo" encodings are deprecated in mbstring. I used the `@` operator on `MailCollector::listEncodings()` to mute deprecation errors, adn deprecate this method as it is not used anymore (I see no occurence on GLPI or on plugins that are available in the marketplace).

3. I replaced a deprecated callable syntax in `Html::cleanPostForTextArea()`.

I was almost able to validate CI run locally, but there are some deprecation errors triggered in following dependencies:
 - `atoum/atoum` (fixed by https://github.com/atoum/atoum/pull/881, PR has to be validated);
 - `mikey179/vfsstream` (fixed by https://github.com/bovigo/vfsStream/commit/17b0f39b4a767bf950066dbad1a9a0b0ec4f7c68, have to be released);
 - `symfony/cache` (Remi is currently preparing a 3.2.0 release of the PHP extension, and we have to wait for it to be able to update `symfony/cache` to latest version, see #10810 );
 - `guzzle/psr7` (see https://github.com/guzzle/psr7/issues/482, I was not able to fix it easilly).

orthagh commented on src/Item_Rack.php:
```diff
@@ -747,9 +751,9 @@ private static function getCell($cell, $readonly = false)
-                        && is_object($item->model)
-                        && isset($item->model->fields['name'])
-                         ? $item->model->fields['name']
+                        && is_object($cell['model'])
```
this change seems strange. Are you sure to move item to cell variable ?

cedric-anne replied:
`$cell` is the parameter that will receive `$outbound` or `$data` items defined here:
![image](https://user-images.githubusercontent.com/33253653/158386168-2c8e96bb-807e-4f30-9148-4590250029a8.png)
![image](https://user-images.githubusercontent.com/33253653/158386191-8df0d787-dd97-42bc-bc72-c3953e53f610.png)

cedric-anne commented:
I added commits to ensure that no BC-break is introduced by private properties declaration.

- Some properties were introduced in GLPI 10.0, so declaring them as private cannot be considered as a BC-break. For them, I did not added any magic logic.
- Some properties are widely used by GLPI and/or plugins, either for read and write operations, so I just declared them as public properties.
- For other properties, I used `__get` and `__set` magic methods to prevent BC-breaks on read/write operations. For those whom I found no usage of them outside corresponding class, I added a deprecation warning.

trasher commented:
Tests are failing

cedric-anne commented:
> Tests are failing

This is due to a weird behaviour (IMHO).
In PHP, if your class declares a property `$a;`, calling `$this->a;` will access this property directly, but if you `unset($this->a);`, then it acts like when the property is not even declared, and it will try to access it through `__get` method.

I fixed it by replacing `unset($this->var)` by `$this->var = null`.
